### DISCUSSION TITLE
How to training on huge text dataset

### DISCUSSION THREAD

**Discussion by Limtle** [Lightning-AI/pytorch-lightning#4139] [DDP / multi-GPU / multi-node]
**What is your question?**

I have a ddp program running on 8 nodes, and I need to load a very huge text dataset ( >30 GB) in this task. However, when I loaded it to dataloader, the program stuck. The intuition is to split my dataset into fractions with smaller size, so every time the dataloader only need to load a small fraction of dataset. Does pytorch-lightning have any support for it? Or any suggestion for solving such problem?

**What's your environment?**
OS: [Linux]
Packaging [pip]
Version [0.10.0]

**Comment by teddykoker** [comment]
Can you tell which part of your script is failing? PyTorch datasets do not need to load all of the data at once if implemented properly, perhaps you can share your `Dataset` implementation?
